### ISSUE TITLE
how to get compact version of `TemporalFusionTransformer`

### ISSUE THREAD

**Issue by turmeric-blend** [jdb78/pytorch-forecasting#305]
Hi, I am interested in `TemporalFusionTransformer`, however, I want to test just the model itself **without** all the additional features like `LightningModule` for training, plotting, logging.. etc

Something like `TemporalFusionTransformer(BaseModelWithCovariates)` with `__init__` and `forward` method only, such that I could run one simple forward pass and get the tensor outputs. Could someone guide me on how to navigate and get a compact version of `TemporalFusionTransformer` to test with?

**Comment by jdb78** [comment]
You can just use the module as it is and treat it as a PyTorch module. It's forward method works as any other PyTorch Module's forward method.

**Comment by turmeric-blend** [comment]
if I am using it as a normal PyTorch Module, do I still need to define the model through `TemporalFusionTransformer.from_dataset(...)`? For example, `from_dataset` parameters like `loss` and `reduce_on_plateau_patience` are usually not defined with the model.

How should I define my model for a normal PyTorch training procedure?

**Comment by turmeric-blend** [comment]
If I want to use a normal PyTroch workflow, would this be the right approach?

```
training = TimeSeriesDataSet(...)
validation = TimeSeriesDataSet.from_dataset(...)

batch_size = 128 
train_dataloader = training.to_dataloader(train=True, batch_size=batch_size, num_workers=0)
val_dataloader = validation.to_dataloader(train=False, batch_size=batch_size * 10, num_workers=0)

loss_fn = ....custom loss fn defined outside "from_dataset"
optimizer = ....custom optimizer fn defined outside "from_dataset"

model = TemporalFusionTransformer.from_dataset(...)

for x, y in train_dataloader:
     model.train()
     output_dict = model(x)
     loss = loss_fn(output_dict['output'], y)
     optimizer.zero_grad()
     loss.backward()
     optimizer.step()
```

**Comment by jdb78** [comment]
Pretty much. You might want to use rescale the predictions with the output_transformer (see the code for the training_step).

**Comment by turmeric-blend** [comment]
I will take a look and reopen if needed. Thanks.

**Comment by TomSteenbergen** [comment]
@turmeric-blend Did you manage to get it to work? If so, could you share your solution? Much appreciated!
